src/MDMTimingCurve.h

```diff
@@ -0,0 +1,152 @@
+/*
+ Copyright 2017-present The Material Motion Authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <CoreGraphics/CoreGraphics.h>
+#import <Foundation/Foundation.h>
+#import <QuartzCore/QuartzCore.h>
+
+/**
+ The possible kinds of traits curves that can be used to describe an animation.
+ */
+typedef NS_ENUM(NSUInteger, MDMTimingCurveType) {
+  /**
+   The value will be instantly set with no animation.
+   */
+  MDMTimingCurveTypeInstant,
+
+  /**
+   The value will be animated using a cubic bezier curve to model its velocity.
+   */
+  MDMTimingCurveTypeBezier,
+
+  /**
+   The value will be animated using a spring simulation.
+
+   A spring will treat the duration property of the traits as a suggestion and may choose to
+   ignore it altogether.
+   */
+  MDMTimingCurveTypeSpring,
+
+} NS_SWIFT_NAME(TimingCurveType);
+
+/**
+ A generalized representation of a traits curve.
+ */
+typedef struct MDMTimingCurve {
+  /**
+   The type defines how to interpret the data values.
+   */
+  MDMTimingCurveType type;
+
+  /**
+   The data values corresponding with this curve.
+   */
+  CGFloat data[4];
```

Comment:
Why do all timing curves have 4*sizeof(CGFloat) worth of data? Should this not be a pointer to another type?

Reply:
Given that these are C-structs, a pointer to another type would be require an extra hurdle to jump over when initializing these values. In practice the timing curves supported by Core Animation only need four pieces of data so this array has been fairly sufficient.

Reply:
Ack.